recipes/netcdf/all/conanfile.py

```diff
@@ -0,0 +1,115 @@
+import os
+import glob
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+
+
+class NetcdfConan(ConanFile):
+    name = "netcdf"
+    description = ("NetCDF is a set of software libraries and "
+                   "self-describing, machine-independent data "
+                   "formats that support the creation, access, "
+                   "and sharing of array-oriented scientific data.")
+    license = "BSD-3-Clause"
+    homepage = "https://www.unidata.ucar.edu/software/netcdf/"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("array", "dataset", "scientific")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_netcdf4": [True, False],
+        "with_dap": [True, False],
+        "with_utilities": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_netcdf4": True,
+        "with_dap": True,
+        "with_utilities": True,
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.with_netcdf4:
+            self.requires("hdf5/[>=1.8.9]")
```

Comment:
Don't use range, set to the latest version available. Why? Reproducibility,

Reply:
I changed the requirements to use the latest available versions.

What I don't understand yet is how this will interplay with new versions of this library. If I add new versions to the `config.yml` in the future which still use the same `conanfile.py` should I then keep these versions as they are as long as that is possible? Or should I update them, which would in turn change how the older versions of this library are built?

Reply:
You can update (manually), since the upstream supports the new version.